### DISCUSSION TITLE
Storing client credentials in a centralized storage

### DISCUSSION THREAD

**Discussion by nermin-pezerovic** [rustdesk/rustdesk#4973] [General]
Hello everyone,

I would like to distribute Rust Desk client to hundreds of machines and i would like to be able to connect to any of them without that user's interaction. I would need to have their ID and their password. I have an idea to have a powershell script that can be distributed using one of our tools to all the machines and that script would be able to pick up passwords and IDs from the machine and push it to our safe centralized storage. Now, with version 1.2.0 of Rust Desk passwords and IDs in RustDesk.toml are hashed and i have no idea how to get the original version. Does anyone have any suggestion of how to achieve this?

**Comment by dinger1986** [comment]
Sounds like a job for RustDesk Pro! Which should be ready for launch soon.

With your tool you can use `--getid` and `--password` with the executable and get the details from there or set the password

**Reply by nermin-pezerovic** [reply]
Is there a way to do it with the free version?

**Reply by dinger1986** [reply]
yeah sorry made the flags more obvious

**Reply by nermin-pezerovic** [reply]
When i run rustdesk.exe for --getid i get an empty line, and when i run rustdesk.exe for --password it just starts the app, but doesn't output any values. Do you have any idea how to work around this?
